schematic/__init__.py

```diff
@@ -95,6 +101,37 @@ def force_flush(self, timeout_millis: int = 30000) -> None:
         pass
 
 
+class CustomFilter(LogRecordProcessor):
+    """A custom log record processor that redacts sensitive data from log messages before they are exported."""
+
+    def __init__(self, exporter):
+        self._exporter = exporter
+        self._shutdown = False
+
+    def redact_google_sheet(self, message: str) -> str:
```

Comment:
It's important we keep the logs when people are running the CLI.  If I'm not mistaken, I think some CLI commands rely on the gsheets link being returned.

My guess is that this would fail some of the CLI tests

Reply:
I don't think this is a concern for CLI usage. The reason is that the log messages that are sent via OTEL is a (deep?) copy of the data sent to stdout/console. 

If I understand this solution it will only affect the messages making their way into SigNoz, but leave them as is in something like cloudwatch. Which, is probably fine since if someone had access to cloud watch, we're already in serious trouble. But, access to SigNoz is going to be given more freely, so we want to curate it carefully.

However, it should be tested to verify these assumption with the cli usage

Reply:
I can't find in the documentation about "deep copy", but my understanding is that open telemetry captures the logs and traces as the application runs independently. Since all the tests passed on Github and locally, I will mark it as "resolved".